fedimint-client-rpc/src/lib.rs

```diff
@@ -119,6 +119,7 @@ pub struct RpcGlobalState {
     clients: Mutex<HashMap<String, ClientHandleArc>>,
     rpc_handles: std::sync::Mutex<HashMap<u64, AbortHandle>>,
     unified_database: Database,
+    preview_cache: Mutex<HashMap<FederationId, ClientPreview>>,
```

Comment:
There are two ways we can go here:
1. What you did: a map with potentially infinite many items. Problem: cache invalidation, increased resource usage, …
2. Instead of a map, just allow caching the last preview using `Mutex<Option<(FederationId, ClientPreview)>>`, avoiding cache invalidation problems, but disallowing use cases like getting a preview for a list of federations while still getting the benefits.

Idk how relevant the "get preview of federation list" use case is, sounds heavy to begin with, but if we needed more info than is available in nostr mint announcements that might be necessary. What's your take @m1sterc001guy?

Reply:
Dev call: it would be best to just use the last previewed federation instead of a full map

Reply:
yes, exactly